presentation-compiler/src/main/dotty/tools/pc/SignatureHelpProvider.scala

```diff
@@ -1,134 +1,149 @@
 
-import scala.jdk.CollectionConverters._
-import scala.meta.pc.OffsetParams
-import scala.meta.pc.SymbolDocumentation
-import scala.meta.pc.SymbolSearch
-
-import dotty.tools.dotc.ast.Trees.AppliedTypeTree
-import dotty.tools.dotc.ast.Trees.TypeApply
-import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.ast.tpd.*
 import dotty.tools.dotc.core.Contexts.*
 import dotty.tools.dotc.core.Flags
 import dotty.tools.dotc.core.Symbols.*
 import dotty.tools.dotc.interactive.Interactive
 import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.parsing.Tokens.closingRegionTokens
+import dotty.tools.dotc.reporting.ErrorMessageID
+import dotty.tools.dotc.reporting.ExpectedTokenButFound
 import dotty.tools.dotc.util.Signatures
-import dotty.tools.dotc.util.Signatures.Signature
 import dotty.tools.dotc.util.SourceFile
-import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.util.Spans
+import dotty.tools.dotc.util.Spans.Span
+import dotty.tools.pc.printer.ShortenedTypePrinter
+import dotty.tools.pc.printer.ShortenedTypePrinter.IncludeDefaultParam
 import dotty.tools.pc.utils.MtagsEnrichments.*
-
 import org.eclipse.lsp4j as l
 
+import scala.jdk.CollectionConverters.*
+import scala.jdk.OptionConverters.*
+import scala.meta.internal.metals.ReportContext
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.SymbolDocumentation
+import scala.meta.pc.SymbolSearch
+
 object SignatureHelpProvider:
 
   def signatureHelp(
       driver: InteractiveDriver,
       params: OffsetParams,
       search: SymbolSearch
-  ) =
-    val uri = params.uri()
-    val sourceFile = SourceFile.virtual(params.uri().nn, params.text().nn)
+  )(using ReportContext): l.SignatureHelp =
+    val uri = params.uri().nn
+    val text = params.text().nn
+    val sourceFile = SourceFile.virtual(uri, text)
     driver.run(uri.nn, sourceFile)
 
-    given ctx: Context = driver.currentCtx
-
-    val pos = driver.sourcePosition(params)
-    val trees = driver.openedTrees(uri.nn)
-
-    val path = Interactive.pathTo(trees, pos)
-
-    val (paramN, callableN, alternatives) =
-      Signatures.signatureHelp(path, pos.span)
-    val infos = alternatives.flatMap { signature =>
-      signature.denot.map {
-        (signature, _)
-      }
-    }
-
-    val signatureInfos = infos.map { case (signature, denot) =>
-      search.symbolDocumentation(denot.symbol) match
-        case Some(doc) =>
-          withDocumentation(
-            doc,
-            signature,
-            denot.symbol.is(Flags.JavaDefined)
-          ).getOrElse(signature)
-        case _ => signature
-
-    }
-
-    /* Versions prior to 3.2.1 did not support type parameters
-     * so we need to skip them.
-     */
-    new l.SignatureHelp(
-      signatureInfos.map(signatureToSignatureInformation).asJava,
-      callableN,
-      paramN
-    )
+    driver.compilationUnits.get(uri) match
+      case Some(unit) =>
+
+        val pos = driver.sourcePosition(params, isZeroExtent = false)
```

Comment:
If you're expanding offset to a range span, won't that throw an error later if the cursor is at the very end of the file? I know, we don't do open brackets but we should make sure not to throw.